contracts/relayer/Relayer.sol

```diff
@@ -0,0 +1,349 @@
+    bytes32 public constant DISALLOW_SENDER_ROLE = keccak256("DISALLOW_SENDER_ROLE");
+
+    // ACL role used to validate who is able to add a new allowed off-chain services to relay transactions
+    bytes32 public constant ALLOW_OFF_CHAIN_SERVICE_ROLE = keccak256("ALLOW_OFF_CHAIN_SERVICE_ROLE");
+
+    // ACL role used to validate who is able to remove already allowed off-chain services to relay transactions
+    bytes32 public constant DISALLOW_OFF_CHAIN_SERVICE_ROLE = keccak256("DISALLOW_OFF_CHAIN_SERVICE_ROLE");
+
+    // ACL role used to validate who is able to change the refunds monthly quota
+    bytes32 public constant SET_MONTHLY_REFUND_QUOTA_ROLE = keccak256("SET_MONTHLY_REFUND_QUOTA_ROLE");
+
+    /**
+     * @dev Event logged when a new address is added to the list of off-chain services allowed to relay transactions
+     * @param service Address of the off-chain service allowed
+     */
+    event ServiceAllowed(address indexed service);
+
+    /**
+     * @dev Event logged when a an address is removed from the list of off-chain services allowed to relay transactions
+     * @param service Address of the off-chain service disallowed
+     */
+    event ServiceDisallowed(address indexed service);
+
+    /**
+     * @dev Event logged when a new address is added to the list of allowed senders to use the relay service
+     * @param sender Address of the sender sallowed to use the relayer service
+     */
+    event SenderAllowed(address indexed sender);
+
+    /**
+     * @dev Event logged when a an address is removed from the list of allowed senders to use the relay service
+     * @param sender Address of the sender disallowed to use the relayer service
+     */
+    event SenderDisallowed(address indexed sender);
+
+    /**
+     * @dev Event logged when a new transaction is relayed successfully
+     * @param from Address executed a transaction on behalf of
+     * @param to Target address of the relayed transaction
+     * @param nonce Nonce of the signer used for the relayed transaction
+     * @param data Calldata included in the relayed transaction
+     */
+    event TransactionRelayed(address from, address to, uint256 nonce, bytes data);
+
+    /**
+     * @dev Event logged when the monthly refunds quota is changed
+     * @param who Address of the account that change the monthly refunds quota
+     * @param previousQuota Previous monthly refunds quota in ETH for each allowed member
+     * @param newQuota New monthly refunds quota in ETH for each allowed member
+     */
+    event MonthlyRefundQuotaSet(address indexed who, uint256 previousQuota, uint256 newQuota);
+
+    // Timestamp to start counting monthly refunds quotas for each member
+    uint256 internal startDate;
+
+    // Monthly refunds quota in ETH for each member
+    uint256 internal monthlyRefundQuota;
+
+    // Mapping that indicates whether a given address is allowed to use the relay service
+    mapping (address => bool) internal allowedSenders;
```

Comment:
Initially we did, but given that using the `ACL` increases gas costs in `~15k`, we decided to use it but in a more "async" way. As you can see, we are using it to edit the whitelist of allowed senders, but avoiding it when we need to check if someone is in included in the list.

Reply:
Ok, then I think this works for services, as I don't expect having so many, but imagine Aragon wants to subsidize some voting (or whatever) to all token holders. According to etherscan, that would be like ~21k holders, i.e., allowed senders. Adding them to that mapping would be quite an amount of gas (and there will probably be bigger orgs with this problem), besides we should monitor ANT transactions to edit allowances.
If we used such an `IGroup` interface, we could have Token Manager implementing it, or if we don't want to update it, just have a wrapper small contract implementing which would get the balance and return true if `> 0`.

Reply:
In that case it would be the same costs even if we use the `ACL` right? I mean, if we want to control each member of a DAO we will need to grant permissions for each of them right?

Reply:
Yea, having groups would be amazing :)